nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestDocumentBlobColdStorage.java

```diff
@@ -0,0 +1,124 @@
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ CoreFeature.class, LogFeature.class, LogCaptureFeature.class })
+@Deploy("org.nuxeo.ecm.core.test.tests:OSGI-INF/test-document-blob-coldstorage-contribs.xml")
+public class TestDocumentBlobColdStorage {
+
+    protected static final String BLOB_FILE_CONTENT = "foo";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Inject
+    protected LogCaptureFeature.Result logCaptureResult;
+
+    @Test
+    public void shouldMoveBlobDocument() throws IOException {
+        DocumentModel documentModel = session.createDocumentModel("/", "AnyFile", "File");
+        documentModel.setPropertyValue("file:content", (Serializable) Blobs.createBlob(BLOB_FILE_CONTENT));
+        documentModel = session.createDocument(documentModel);
+        assertFalse(documentModel.hasFacet(COLD_STORAGE));
+
+        // move the blob to cold storage
+        moveBlob(session, documentModel.getRef());
+        session.saveDocument(documentModel);
+        transactionalFeature.nextTransaction();
+
+        documentModel = session.getDocument(documentModel.getRef());
+        assertTrue(documentModel.hasFacet(COLD_STORAGE));
+
+        // check if the `file:content` contains the thumbnail blob
+        checkBlobContent(documentModel, FILE_CONTENT_PROPERTY, FAKE_THUMBNAIL_CONTENT);
+
+        // check if the `coldstorage:coldContent` contains the original file content
+        checkBlobContent(documentModel, COLD_STORAGE_CONTENT_PROPERTY, BLOB_FILE_CONTENT);
+    }
+
+    @Test
+    @LogCaptureFeature.FilterOn(loggerClass = DocumentBlobColdStorageHelper.class, logLevel = "TRACE")
+    @LoggerLevel(klass = DocumentBlobColdStorageHelper.class, level = "TRACE")
+    public void shouldLogMessageWhenMovingDocumentBlobAlreadyOnColdStorage() {
+        DocumentModel documentModel = session.createDocumentModel("/", "MyFile", "File");
+        documentModel.setPropertyValue("file:content", (Serializable) Blobs.createBlob(BLOB_FILE_CONTENT));
+        documentModel = session.createDocument(documentModel);
+        assertFalse(documentModel.hasFacet(COLD_STORAGE));
+
+        // move the first time
+        moveBlob(session, documentModel.getRef());
+        session.saveDocument(documentModel);
+        transactionalFeature.nextTransaction();
+        documentModel = session.getDocument(documentModel.getRef());
+        assertTrue(documentModel.hasFacet(COLD_STORAGE));
+
+        // make another move
+        moveBlob(session, documentModel.getRef());
+        List<String> caughtEvents = logCaptureResult.getCaughtEventMessages();
+        assertEquals(1, caughtEvents.size());
+        assertEquals(String.format("The blob content of document: %s is already on cold storage", documentModel),
+                caughtEvents.get(0));
+    }
+
+    protected void checkBlobContent(DocumentModel documentModel, String xpath, String expectedContent)
```

Comment:
You can simply use `((Blob) documentModel.getPropertyValue(xpath)).getString()`, directly in the test.

In fact, to enforce the check, maybe:
```
Blob blob content = (Blob) documentModel.getPropertyValue(xpath);
assertNotNull(content);
assertEquals("XYZ", content.getString())
```

But I would still keep these assertions inside the test, for readability.